### PR TITLE
chore: Implemented the internal session payments call and payment details and the session setup call

### PR DESCRIPTION
**Pull request by mohammedDehairy**
chore: Implemented the internal session payments call and payment details and the session setup call

### CODE REVIEW

**Review by erenbesel — DISMISSED** [review]

File: AdyenSession/Session.swift [hunk 1]
```diff
@@ -8,28 +8,104 @@ import Adyen
...
-    private init(sessionData: String) {
-        self.sessionData = sessionData
+    /// The session information
+    public struct Context {
+        
+        /// The session data.
+        public internal(set) var data: String
+        
+        /// The session idenitifier
+        public let idenitifier: String
```
Comment (nauaros) [inline AdyenSession/Session.swift]: Is this a typo?

File: AdyenSession/Session.swift [hunk 2]
```diff
@@ -8,28 +8,104 @@ import Adyen
...
 import Foundation
 
 /// The Session object.
 public final class Session: SessionProtocol {
     
-    private let sessionData: String
+    /// Session configuration.
+    public struct Configuration {
+        
+        internal let sessionIdenitifier: String
```
Comment (nauaros) [inline AdyenSession/Session.swift]: Also typo here

File: AdyenSession/API/Partial Payments API/Create Order/CreateOrderRequest.swift [hunk 3]
```diff
@@ -0,0 +1,63 @@
...
+
+    internal let headers: [String: String] = [:]
+
+    internal let queryParameters: [URLQueryItem] = []
+
+    internal let sessionData: String
+    
+    internal init(sessionId: String,
+                  sessionData: String) {
+        self.path = "checkoutshopper/v1/sessions/\(sessionId)/orders"
```
Comment (erenbesel) [inline AdyenSession/API/Partial Payments API/Create Order/CreateOrderRequest.swift:29]: Shouldn't we have this `checkoutshopper/v1/sessions/` part as a constant? Would be annoying to change each one when there is a version upgrade

Comment (mohammedDehairy) [reply AdyenSession/API/Partial Payments API/Create Order/CreateOrderRequest.swift:29]: can I fix this in an another PR?, its used in all other checkoutshopper requests

**Review by erenbesel — APPROVED** [review]

**Review by nauaros — APPROVED** [review]

File: Adyen.xcodeproj/project.pbxproj [hunk 4]
```diff
@@ -6532,7 +6640,7 @@
 				GENERATE_INFOPLIST_FILE = YES;
 				INFOPLIST_KEY_NSHumanReadableCopyright = "Copyright © 2022 Adyen. All rights reserved.";
 				INSTALL_PATH = "$(LOCAL_LIBRARY_DIR)/Frameworks";
-				IPHONEOS_DEPLOYMENT_TARGET = 15.2;
+				IPHONEOS_DEPLOYMENT_TARGET = 11.0;
```
Comment (descorp) [inline Adyen.xcodeproj/project.pbxproj:6643]: this is important 😆

Comment (mohammedDehairy) [reply Adyen.xcodeproj/project.pbxproj:6643]: hahahaha

**Pull request merged**